src/cli/tc.rs

```diff
@@ -22,6 +23,7 @@ const FILE_NAME: &str = ".tailcallrc.graphql";
 const YML_FILE_NAME: &str = ".graphqlrc.yml";
 
 pub async fn run() -> Result<()> {
+    let _ = dotenv(); // dotenvy fails if .env is not found, but we don't care if it doesn't exist.
```

Comment:
> - this reads sync, we need an asynchronous solution

Reading a miniscule `.env` file once at the start of a CLI run should not *need* to be async, especially since you can't really do anything else in the "time" it would take to read and parse the file.

> - this reads from the path where the `cargo` is running, not from where the config itself is declared

You can use `dotenvy::from_path`.

I won't drive this PR to completion since you have assignment, but you should consider this instead of rolling your own async env reader for not much benefit.